File: components.py
class Components:
    def __init__(self,n):
        # TODO

    def add_road(self,a,b):
        # TODO

    def count(self):
        # TODO

if __name__ == "__main__":
    c = Components(5)
    print(c.count()) # 5
    c.add_road(1,2)
    c.add_road(1,3)
    print(c.count()) # 3
    c.add_road(2,3)
    print(c.count()) # 3
    c.add_road(4,5)
    print(c.count()) # 2

File: Java_template/Components.java
public class Components {
    public Components(int n) {
        // TODO
    }

    public void addRoad(int a, int b) {
        // TODO
    }

    public int count() {
        // TODO
    }

    public static void main(String[] args) {
        Components c = new Components(5);
        System.out.println(c.count()); // 5
        c.addRoad(1,2);
        c.addRoad(1,3);
        System.out.println(c.count()); // 3
        c.addRoad(2,3);
        System.out.println(c.count()); // 3
        c.addRoad(4,5);
        System.out.println(c.count()); // 2
    }
}
